### PR TITLE
Adding support for different target host than the executing environment

### LINKED ISSUE #254: Support other GitHub instance then the current one in the executing workflow
## Problem Statement
We use this on action on GitHub Enterprise Server to fork actions on github.com. Since the action runs in the local context by default, we need to make some changes to support this.

## Proposed Solution

Add a parameter for the target instance url

## Alternatives Considered
- none

## Additional Context
We already use the action with these changes, so I will contribute them back here as well.

### PR DESCRIPTION
**Pull request by rajbos**
## Description

See the description in #254 as well. We use this on action on our private GitHub Enterprise Server to fork actions on github.com. Since the action runs in the local context by default, we need to make some changes to support this. This PR adds a new optional parameter to support targeting a different environment. If the variable is empty, the current behavior is still the same: it uses the current environment from `actions/core`.

If the new parameter `targetInstanceUrl` is set, it will configure Octokit with that parameter. The rest of the code is the same.

Fixes #254 

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/forker/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests  ➡️ cannot be done from a github.com environment, as there is nothing else to target.
- [x] All new and existing tests pass

Tagging @lelia for visibility 😄.

### CODE REVIEW

**Comment by rajbos** [comment]
@lelia anything I can do to help with this? I can confirm it works as intended on our internal environment 🤗.

**Review by lelia — CHANGES_REQUESTED** [review]
@rajbos Sorry for the delay here, I haven't worked at Wayfair since the start of the year, so my primary focus has been on other OSS projects at my new company 😔

While I haven't had a chance to explicitly test out the parameter using a GHE instance yet, the overall changes LGTM. My one ask would be that you document the new parameter in the `README` [Inputs](https://github.com/wayfair-incubator/forker#inputs) section following the same pattern as other params (so, datatype, required/not, description, and example input). 

Bonus points if you expand the [Usage](https://github.com/wayfair-incubator/forker#usage) section to include a code snippet for how one would construct this action to target a dummy GHE instance. Thanks and apologies again for the wait!

**Comment by lelia** [comment]
BTW, feel free to update this PR description to say something like "Fixes #254" so that the related issue will auto-close once this PR merges!

**Comment by rajbos** [comment]
@lelia : requested changes have been added, please review.

File: README.md [hunk 1]
```diff
@@ -136,6 +142,19 @@ with:
   licenseAllowlist: "0bsd\napache-2.0\nmit"
 ```
 
+### Target a GitHub Enterprise Server
+Create a fork in your own GitHub Enterprise Server:
```
Comment (lelia) [inline README.md:146]:
```suggestion
### Target a GitHub Enterprise Server

Create a fork in your own GitHub Enterprise Server:
```

**Review by lelia — APPROVED** [review]
Changes LGTM! Thanks again for the contribution & your patience 🙂

File: README.md [hunk 2]
```diff
@@ -81,6 +81,12 @@ A newline-delimited (`"\n"`) string representing a list of allowed [license keys
 
 **Example:** `"0bsd\napache-2.0\nmit"`
 
+### `targetInstanceUrl` (optional, string)
+
+This parameter can be used to target a GitHub Enterprise Server instance instead of https://github.com.
```
Comment (lelia) [inline README.md:86]:
```suggestion
This parameter can be used to target a GitHub Enterprise Server instance instead of <https://github.com>.
```

**Pull request merged**